### PR TITLE
GEOPY-1497: fix reduced set of linter errors on simpeg-drivers

### PR DESCRIPTION
**Pull request by sebhmg**
**GEOPY-1497 - fix reduced set of linter errors on simpeg-drivers**

### CODE REVIEW

File: simpeg_drivers/electricals/direct_current/two_dimensions/__init__.py [hunk 1]
```diff
@@ -15,5 +15,6 @@
 #
 # ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
 
-
 from .params import DirectCurrent2DParams
+
+__all__ = ["DirectCurrent2DParams"]
```
Comment (sebhmg) [inline simpeg_drivers/electricals/direct_current/two_dimensions/__init__.py:20]: note: we should actually do that in every `__init__.py`
(and not ignore flake8)

I did not know about `__all__` when we initiated the work on geoh5py

File: simpeg_drivers/components/data.py [hunk 2]
```diff
@@ -171,14 +171,14 @@ def drape_locations(self, locations: np.ndarray) -> np.ndarray:
 
         return np.c_[distance_interp, locations[:, 2:]]
 
-    def filter(self, a):
+    def filter(self, obj: dict[str, np.ndarray] | np.ndarray, mask=None):
         """Remove vertices based on mask property."""
         if self.indices is None:
-            self.indices = np.where(self.mask)[0]
+            self.indices = np.where(mask)[0]
```
Comment (domfournier) [inline simpeg_drivers/components/data.py:177]: That doesn't work.

Comment (sebhmg) [reply simpeg_drivers/components/data.py:177]: ok, @benk-mira , can you take it from here?
this change is coming from https://github.com/MiraGeoscience/simpeg-drivers/pull/23

Comment (sebhmg) [reply simpeg_drivers/components/data.py:181]: ok, brought in changes from https://github.com/MiraGeoscience/simpeg-drivers/pull/19 ( GEOPY-1477 )

**Review by domfournier — APPROVED** [review]

**Pull request merged**